processor/error/processor.go

```diff
@@ -20,12 +20,12 @@ const (
 func NewProcessor() pr.Processor {
 	schema := pr.CreateSchema(errorSchema, processorName)
 	return &processor{
-		&Payload{},
+		&payload{},
 		schema}
 }
 
 type processor struct {
-	payload *Payload
+	payload pr.Payload
```

Comment:
Yes,Changing this to the local payload looks better than the interface.Will change it.